### PR TITLE
feat: support using mcp server on sse mode using GCP_STDIO

### PR DESCRIPTION
**Pull request by shruti-mantri**
This is based on the observations here: https://github.com/GoogleCloudPlatform/cloud-run-mcp/issues/113
Closes https://github.com/GoogleCloudPlatform/cloud-run-mcp/issues/113

Ideally, setting GCP_STDIO as false should start the server in SSE mode, which was not the case.
This PR ensures this behavior.

### CODE REVIEW

**Review by steren — CHANGES_REQUESTED** [review]
Thanks for the change, LGTM, but this is not something we should change without warning in the future, so I think we should add tests for this behavior.  Can you add some?

File: test/local/mcp-server-streamable-http.test.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    });
+
+    after(async () => {
+        await client.cleanup();
+        if (serverProcess) {
+            serverProcess.kill();
+        }
+    });
+
+    test('should list tools', async () => {
```
Comment (steren) [inline test/local/mcp-server-streamable-http.test.js]: I think we already test for that in https://github.com/GoogleCloudPlatform/cloud-run-mcp/blob/main/test/local/tools.test.js

What this test file should focus on is the behavior of the MCP server when it comes to the presence of GCP_STDIO envvar or detection of GCP project, where it starts or not the stdio transport

**Review by steren — CHANGES_REQUESTED** [review]

File: test/local/mcp-server-streamable-http.test.js (hunk 1 above)
Comment (shruti-mantri) [reply test/local/mcp-server-streamable-http.test.js]: I agree. But this would require more refactoring, wherein all the logic will be moved to a method which can then be mocked. I am planning to take this up as a separate PR.

The present change is not changing the existing behavior, but adding to it. GCP_STDIO was as good as non-functional until now. With this change, the env var is actually taking effect. So, this change is good to go in for now.

Comment (steren) [reply test/local/mcp-server-streamable-http.test.js]: OK, then:
- please open a new tracking issue for adding tests for the studio / httpstreaming behavior
- delete this test file, as it is not testing more than what is already tested in existing test files

Comment (shruti-mantri) [reply test/local/mcp-server-streamable-http.test.js]: - Created the new ticket for adding the test case: https://github.com/GoogleCloudPlatform/cloud-run-mcp/issues/128
- The added test file checks if the local server when started with `GCP_STDIO=false` indeed starts the streamable-http server, and the tools and prompts are listed as expected. We should keep this test to ensure that local hosting  of streamable-http server works as expected. This test is equivalent to what mcp-server.test.js is testing for stdio mode, but for streamble-http.

Comment (steren) [reply test/local/mcp-server-streamable-http.test.js]: Can you update this test file?
I expect to see it('should start an http server') if what it is testing is that an http server is started.

We should test that by sending an HTTP request to the opened port on localhost.

Listing the tools and prompts are already tested in another test files, we do not need to duplicate this test here.

**Review by steren — CHANGES_REQUESTED** [review]

**Comment by shruti-mantri** [comment]
The test case is testing exactly that!
If you look closely, the client is started with the provided url in its constructor:
this.client = new Client({ name: `mcp-client-for-${serverName}`, version: "1.0.0", url: `http://localhost:3000/mcp` });

The tools  are listed with the help of this client which is connecting to the locally hosted server in the `before` section.
I will reword the test case description to highlight that this is for streamble-http.

**Comment by shruti-mantri** [comment]
Done
